packages/api/lib/parse-pubsub-payload.js

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+const assert = require('assert');
+const firebaseEncode = require('./firebase-encode');
+
+function buildInfo (metadata) {
+  const [org, repo] = metadata.repo.split('/');
+  const match = metadata.buildURL.match(/\[Sponge\]\((?<url>.*)\)/);
+  return {
+    repoId: firebaseEncode(decodeURIComponent(metadata.repo)),
+    organization: org,
+    timestamp: new Date(),
+    url: match ? match.groups.url : `https://github.com/${metadata.repo}`,
+    environment: {
+      matrix: '{"kokoro":""}',
+      os: 'Linux',
+      ref: metadata.commit,
+      tag: 'None'
+    },
+    buildId: match ? match.groups.url : metadata.commit,
+    sha: metadata.commit,
+    name: repo,
+    description: '',
+    buildmessage: ''
+  };
+}
+
+module.exports = (pubsubPayload) => {
+  assert(pubsubPayload.message, 'pubsub payload did not contain message field');
```

Comment:
I've never seen assert used outside of a test class like this. I'm curious, what does this do? It seems to *require* that the message fields be left empty?

Reply:
@cedpeters as in tests, it will throw an exception if the statement doesn't evaluate to `true`.

It can be a good way to make an error related to inputs more clear, otherwise the exception might occur later in the program because a field can't be found on an object you were  expecting.

Reply:
@bcoe Ah okay, thank you! I'm curious why we specifically demand that the pubsub must not contain a message field. Would we run into problems if we just asserted that the message has to exist, and not that it necessarily must be that specific one?